### PR TITLE
feat(txfees/incentives): minimum base fee for a message

### PR DESCRIPTION
**Pull request by p0mvn**
Closes: #XXX

## What is the purpose of the change

Adds minimum fee for certain messages by creating a `MsgMinFeeExtension` interface. By extending this interface extension, messages return a min fee in the base denom.

The fee decorator then extracts this fee from every tx and compares it to the gas fee. The maximum of min gas fee and min message is compared to the actual transaction fee to determine if a transaction is valid.

## Brief Changelog

- create the interface
- extend the interface from `CreateGauge` message and `AddToGauge` message
- wire feedocrator
- add a test

## Testing and Verifying

This change added tests and can be verified as follows:
- `TestIsSufficientFee`

## Documentation and Release Note

  - Does this pull request introduce a new feature or user-facing behavior changes? yes
  - Is a relevant changelog entry added to the `Unreleased` section in `CHANGELOG.md`? yes
  - How is the feature or change documented? not applicable

### CODE REVIEW

File: x/incentives/types/msgs.go [hunk 1]
```diff
@@ -5,16 +5,22 @@ import (
...
 
 	sdk "github.com/cosmos/cosmos-sdk/types"
 )
 
 const (
 	TypeMsgCreateGauge = "create_gauge"
 	TypeMsgAddToGauge  = "add_to_gauge"
+
+	createGaugeMinBaseFee = 2
+	addToGaugeMinBaseFee  = 1
```
Comment (p0mvn) [inline x/incentives/types/msgs.go:18]: Looking for suggestions on the values here please

@ValarDragon @czarcas7ic

Comment (czarcas7ic) [reply x/incentives/types/msgs.go:18]: We can do 50 OSMO for create and 25 for add for now I think

File: x/incentives/types/msgs.go [hunk 2]
```diff
@@ -5,16 +5,22 @@ import (
...
 	TypeMsgCreateGauge = "create_gauge"
 	TypeMsgAddToGauge  = "add_to_gauge"
+
+	createGaugeMinBaseFee = 50 * 1_000_000
+	addToGaugeMinBaseFee  = 25 * 1_000_000
 )
 
 var _ sdk.Msg = &MsgCreateGauge{}
+var _ txfeestypes.MsgMinFeeExtension = &MsgCreateGauge{}
+var _ txfeestypes.MsgMinFeeExtension = &MsgAddToGauge{}
```
Comment (alexanderbez) [inline x/incentives/types/msgs.go:23]: nit: can we group all these?

File: x/txfees/keeper/feedecorator.go [hunk 3]
```diff
@@ -92,23 +93,39 @@ func (mfd MempoolFeeDecorator) AnteHandle(ctx sdk.Context, tx sdk.Tx, simulate b
 	return next(ctx, tx, simulate)
 }
 
-// IsSufficientFee checks if the feeCoin provided (in any asset), is worth enough osmo at current spot prices
-// to pay the gas cost of this tx.
-func (k Keeper) IsSufficientFee(ctx sdk.Context, minBaseGasPrice sdk.Dec, gasRequested uint64, feeCoin sdk.Coin) error {
+// IsSufficientFee checks if the tx provided (in any asset), is worth enough osmo at current spot prices
```
Comment (alexanderbez) [inline x/txfees/keeper/feedecorator.go:96]:
```suggestion
// IsSufficientFee checks if the fee provided (in any asset), is worth enough osmo at current spot prices
```

File: x/txfees/keeper/feedecorator.go [hunk 4]
```diff
@@ -92,23 +93,39 @@ func (mfd MempoolFeeDecorator) AnteHandle(ctx sdk.Context, tx sdk.Tx, simulate b
...
+// - gas cost of this tx
+// - minimum fee associated with any of the messages in this tx. N.B.: not all messages have such a fee.
+func (k Keeper) IsSufficientFee(ctx sdk.Context, minBaseGasPrice sdk.Dec, tx sdk.FeeTx) error {
 	baseDenom, err := k.GetBaseDenom(ctx)
 	if err != nil {
 		return err
 	}
+	feeCoins := tx.GetFee()
+	if len(feeCoins) != 1 {
+		return errors.Wrap(sdkerrors.ErrInsufficientFee, fmt.Sprintf("need one fee coin, got %d, %s", len(feeCoins), feeCoins))
```
Comment (alexanderbez) [inline x/txfees/keeper/feedecorator.go:107]:
```suggestion
		return errors.Wrap(sdkerrors.ErrInsufficientFee, fmt.Sprintf("invalid number of fee tokens provided (%s); expected: 1, got: %d", feeCoins, len(feeCoins)))
```

**Review by alexanderbez — APPROVED** [review]
Looks sound to me!

**Review by czarcas7ic — APPROVED** [review]
LGTM 👍

File: x/incentives/types/msgs.go [hunk 5]
```diff
@@ -5,16 +5,24 @@ import (
...
 
 	sdk "github.com/cosmos/cosmos-sdk/types"
 )
 
 const (
 	TypeMsgCreateGauge = "create_gauge"
 	TypeMsgAddToGauge  = "add_to_gauge"
+
+	createGaugeMinBaseFee = 50 * 1_000_000
+	addToGaugeMinBaseFee  = 25 * 1_000_000
```
Comment (stackman27) [inline x/incentives/types/msgs.go:18]: how were these values decided?

Comment (czarcas7ic) [reply x/incentives/types/msgs.go:18]: Do you feel they are too high or too low?

**Review by mattverse — COMMENTED** [review]
ACK,

I feel like it be best if these are params just as pool creation fee

**Pull request merged**